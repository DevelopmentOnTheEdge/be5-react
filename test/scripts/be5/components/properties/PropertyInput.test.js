import React from 'react';
import PropertyInput from '../../../../../src/scripts/be5/components/properties/propertyInput';
import renderer from 'react-test-renderer';
/*
    contained values for input
*/

it('input', () => {
    const bean = require('./testJson.json');

    const handle = jest.fn();

    let component = renderer.create(
        <PropertyInput path={"/textInput"} bean={bean} onChange={handle} />
    );

    expect(component.toJSON()).toMatchSnapshot();

    bean.values.textInput = "";

    component = renderer.create(
        <PropertyInput path={"/textInput"} bean={bean} onChange={handle} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});

/*
  tests for select
*/

it('select', () => {
    const bean = require('./testJson.json');

    const handle = jest.fn();

    let component = renderer.create(
        <PropertyInput path={"/select"} bean={bean} onChange={handle} />
    );

    expect(component.toJSON()).toMatchSnapshot();

    //check for null values
    bean.values.select = "";
    component = renderer.create(
        <PropertyInput path={"/select"} bean={bean} onChange={handle} />
    );

    expect(component.toJSON()).toMatchSnapshot();

    //check for valid values
    bean.values.select = "watermelon";
    component = renderer.create(
        <PropertyInput path={"/select"} bean={bean} onChange={handle} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});

/*
  tests for multiSelect
*/

it('multiSelect', () => {
    const bean = require('./testJson.json');

    const handle = jest.fn();

    let component = renderer.create(
        <PropertyInput path={"/multiSelect"} bean={bean} onChange={handle} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});

/*
  tests for description
*/

it('Description', () => {
    const bean = require('./testJson.json');

    const handle = jest.fn();

    let component = renderer.create(
        <PropertyInput path={"/description"} bean={bean} onChange={handle} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});

/*
  tests for passwordField
*/

it('passwordField', () => {
    const bean = require('./testJson.json');

    const handle = jest.fn();

    let component = renderer.create(
        <PropertyInput path={"/pass"} bean={bean} onChange={handle} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});

/*
  tests for login
*/

it('property login', () => {
    const bean = require('./testJson.json');

    const handle = jest.fn();

    let component = renderer.create(
        <PropertyInput path={"/login"} bean={bean} onChange={handle} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});

it('property Boolean', () => {
    const bean = require('./testJson.json');

    const handle = jest.fn();

    let component = renderer.create(
        <PropertyInput path={"/agree"} bean={bean} onChange={handle} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});

/*
  tests for lableField
*/

it('property labelField', () => {
    const bean = require('./testJson.json');

    const handle = jest.fn();

    let component = renderer.create(
        <PropertyInput path={"/label"} bean={bean} onChange={handle} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});

//to do: не работает, выдает непонятную ошибку
// it('property data', () => {
//     const bean = require('./testJson.json');
//
//     const handle = jest.fn();
//
//     let component = renderer.create(
//         <PropertyInput path={"/date"} bean={bean} onChange={handle} />
//     );
//
//     expect(component.toJSON()).toMatchSnapshot();
// });
