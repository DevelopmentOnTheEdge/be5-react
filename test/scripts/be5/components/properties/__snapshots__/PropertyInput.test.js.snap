// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description 1`] = `
<textarea
  className="form-control"
  cols={undefined}
  disabled={undefined}
  id="descriptionField"
  onChange={[Function]}
  placeholder="field for text"
  rows={3}
  value=""
/>
`;

exports[`input 1`] = `
<input
  className="form-control"
  disabled={undefined}
  id="textInputField"
  onChange={[Function]}
  placeholder={undefined}
  type="text"
  value="text"
/>
`;

exports[`input 2`] = `
<input
  className="form-control"
  disabled={undefined}
  id="textInputField"
  onChange={[Function]}
  placeholder={undefined}
  type="text"
  value=""
/>
`;

exports[`multiSelect 1`] = `
<div
  className="Select Select--multi is-clearable is-searchable"
  style={undefined}
>
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <span
      className="Select-multi-value-wrapper"
      id="react-select-5--value"
    >
      <div
        className="Select-placeholder"
      >
        Select ...
      </div>
      <div
        className="Select-input"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <input
          aria-activedescendant="react-select-5--value"
          aria-describedby={undefined}
          aria-expanded="false"
          aria-haspopup="false"
          aria-label={undefined}
          aria-labelledby={undefined}
          aria-owns=""
          className={undefined}
          id={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          role="combobox"
          style={
            Object {
              "boxSizing": "content-box",
              "width": "5px",
            }
          }
          tabIndex={undefined}
          value=""
        />
        <div
          style={
            Object {
              "height": 0,
              "left": 0,
              "overflow": "scroll",
              "position": "absolute",
              "top": 0,
              "visibility": "hidden",
              "whiteSpace": "pre",
            }
          }
        >
          
        </div>
      </div>
    </span>
    <span
      aria-label="Clear all"
      className="Select-clear-zone"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      title="Clear all"
    >
      <span
        className="Select-clear"
        dangerouslySetInnerHTML={
          Object {
            "__html": "&times;",
          }
        }
      />
    </span>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="Select-arrow"
        onMouseDown={[Function]}
      />
    </span>
  </div>
</div>
`;

exports[`passwordField 1`] = `
<input
  className="form-control"
  disabled={undefined}
  id="passField"
  onChange={[Function]}
  placeholder={undefined}
  type="password"
  value="password"
/>
`;

exports[`property Boolean 1`] = `
<input
  checked={true}
  className="form-check-input"
  disabled={undefined}
  id="agreeField"
  onChange={[Function]}
  type="checkbox"
  value={true}
/>
`;

exports[`property labelField 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p><strong>Pellentesque habitant morbi tristique</strong> senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. <em>Aenean ultricies mi vitae est.</em> Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, <code>commodo vitae</code>, ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. <a href=\\"#\\">Donec non enim</a> in turpis pulvinar facilisis. Ut felis.</p><ol><li>Lorem ipsum dolor sit amet, consectetuer adipiscing elit.</li>   <li>Aliquam tincidunt mauris eu risus.</li></ol>",
    }
  }
/>
`;

exports[`property login 1`] = `
<input
  className="form-control"
  disabled={undefined}
  id="loginField"
  onChange={[Function]}
  placeholder={undefined}
  type="text"
  value=""
/>
`;

exports[`property maskTest 1`] = `
<input
  className="form-control"
  disabled={undefined}
  id="maskInputField"
  onChange={[Function]}
  placeholder={undefined}
  type="text"
  value=""
/>
`;

exports[`select 1`] = `
<div
  className="Select Select--single is-clearable is-searchable has-value"
  style={undefined}
>
  <input
    disabled={false}
    name="selectField"
    type="hidden"
    value="vanilla"
  />
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <span
      className="Select-multi-value-wrapper"
      id="react-select-2--value"
    >
      <div
        className="Select-value"
        style={undefined}
        title={undefined}
      >
        <span
          aria-selected="true"
          className="Select-value-label"
          id="react-select-2--value-item"
          role="option"
        >
          Vanilla
        </span>
      </div>
      <div
        className="Select-input"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <input
          aria-activedescendant="react-select-2--value"
          aria-describedby={undefined}
          aria-expanded="false"
          aria-haspopup="false"
          aria-label={undefined}
          aria-labelledby={undefined}
          aria-owns=""
          className={undefined}
          id={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          role="combobox"
          style={
            Object {
              "boxSizing": "content-box",
              "width": "5px",
            }
          }
          tabIndex={undefined}
          value=""
        />
        <div
          style={
            Object {
              "height": 0,
              "left": 0,
              "overflow": "scroll",
              "position": "absolute",
              "top": 0,
              "visibility": "hidden",
              "whiteSpace": "pre",
            }
          }
        >
          
        </div>
      </div>
    </span>
    <span
      aria-label="Clear value"
      className="Select-clear-zone"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      title="Clear value"
    >
      <span
        className="Select-clear"
        dangerouslySetInnerHTML={
          Object {
            "__html": "&times;",
          }
        }
      />
    </span>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="Select-arrow"
        onMouseDown={[Function]}
      />
    </span>
  </div>
</div>
`;

exports[`select 2`] = `
<div
  className="Select Select--single is-clearable is-searchable"
  style={undefined}
>
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <span
      className="Select-multi-value-wrapper"
      id="react-select-3--value"
    >
      <div
        className="Select-placeholder"
      >
        Select ...
      </div>
      <div
        className="Select-input"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <input
          aria-activedescendant="react-select-3--value"
          aria-describedby={undefined}
          aria-expanded="false"
          aria-haspopup="false"
          aria-label={undefined}
          aria-labelledby={undefined}
          aria-owns=""
          className={undefined}
          id={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          role="combobox"
          style={
            Object {
              "boxSizing": "content-box",
              "width": "5px",
            }
          }
          tabIndex={undefined}
          value=""
        />
        <div
          style={
            Object {
              "height": 0,
              "left": 0,
              "overflow": "scroll",
              "position": "absolute",
              "top": 0,
              "visibility": "hidden",
              "whiteSpace": "pre",
            }
          }
        >
          
        </div>
      </div>
    </span>
    <span
      aria-label="Clear value"
      className="Select-clear-zone"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      title="Clear value"
    >
      <span
        className="Select-clear"
        dangerouslySetInnerHTML={
          Object {
            "__html": "&times;",
          }
        }
      />
    </span>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="Select-arrow"
        onMouseDown={[Function]}
      />
    </span>
  </div>
</div>
`;

exports[`select 3`] = `
<div
  className="Select Select--single is-clearable is-searchable"
  style={undefined}
>
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <span
      className="Select-multi-value-wrapper"
      id="react-select-4--value"
    >
      <div
        className="Select-placeholder"
      >
        Select ...
      </div>
      <div
        className="Select-input"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <input
          aria-activedescendant="react-select-4--value"
          aria-describedby={undefined}
          aria-expanded="false"
          aria-haspopup="false"
          aria-label={undefined}
          aria-labelledby={undefined}
          aria-owns=""
          className={undefined}
          id={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          role="combobox"
          style={
            Object {
              "boxSizing": "content-box",
              "width": "5px",
            }
          }
          tabIndex={undefined}
          value=""
        />
        <div
          style={
            Object {
              "height": 0,
              "left": 0,
              "overflow": "scroll",
              "position": "absolute",
              "top": 0,
              "visibility": "hidden",
              "whiteSpace": "pre",
            }
          }
        >
          
        </div>
      </div>
    </span>
    <span
      aria-label="Clear value"
      className="Select-clear-zone"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      title="Clear value"
    >
      <span
        className="Select-clear"
        dangerouslySetInnerHTML={
          Object {
            "__html": "&times;",
          }
        }
      />
    </span>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="Select-arrow"
        onMouseDown={[Function]}
      />
    </span>
  </div>
</div>
`;
